### PR TITLE
fix-#392: added related blogs section in detailed blog page

### LINKED ISSUE #392: [BUG] additional card in detail page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

in detail page nothing to much 
look like a boring also short information

### Expected Behavior

in detail page ,
we can add the additional card like tourism website or flipkart

for example :we click on shoes then display the shoes related information additionally other shoes also display for user visit more and more website

### Reproduction steps

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```

### Screenshots

DESCRIPTION
image
like this :
image

### Logs

_No response_

### Browsers

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by vamsidhar-914**
## Summary

added  related blogs section in detailed blog page

## Description

whenever user visited individual blog page..they can see related posts based on the categories in the blog page

## Images
desktop view
Screenshot (200)

mobile view
Screenshot (201)

## Issue(s) Addressed

Closes #392 

## Prerequisites

- [ yes] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

**Comment by vamsidhar-914** [comment]
hey @krishnaacharyaa please do review it and want any changes then tell me 👍

**Comment by vamsidhar-914** [comment]
helllo @krishnaacharyaa ??

File: frontend/src/pages/details-page.tsx [hunk 1]
```diff
@@ -30,6 +37,24 @@ export default function DetailsPage() {
     }
   }, [post]);
 
+  const category = post.categories[0];
+  useEffect(() => {
```
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:41]: It should use all the categories, 
Basically if there are three categories a b and c , i need to get all the posts which has any of a b or c. and then get only top 3 posts in the mobile view, vertically and may be 4 or 5 in the desktop view

File: frontend/src/pages/details-page.tsx [hunk 2]
```diff
@@ -12,6 +17,8 @@ export default function DetailsPage() {
   const [loading, setIsLoading] = useState(initialVal);
   const { postId } = useParams();
   const navigate = useNavigate();
+  const [categoryPosts, setcategoryPosts] = useState<Post[]>([]);
+  const [postLoading, setpostLoading] = useState(false);
```
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:22]: I see this is declared, but never used anywhere, ideally this should be used and the naming should be relatedPostsLoading and shimmer view needs to be provided when it is loading

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
Thank you for the PR @vamsidhar-914 kindly address the review comments

**Comment by vamsidhar-914** [comment]
ok sure @krishnaacharyaa on itt

**Comment by vamsidhar-914** [comment]
hi @krishnaacharyaa 
 
fetched all related category posts and displayed 3 posts in mobile view and 4 posts in desktop view

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
I really appreciate the efforts, but we have a little of tasks to be done, kindly address the review comments

**Comment by krishnaacharyaa** [comment]
For mobile can we aim for this kind (style) of related posts? 
image

small icon in the left and then the content on the right.

If you feel these are quite a few requests in a single PR, kindly let me know we can break this into subtasks, and we can raise this as another PR :) I'm totally fine with that

**Comment by vamsidhar-914** [comment]
no its finee...i'll be on it

**Comment by vamsidhar-914** [comment]
@krishnaacharyaa 

changed naming conventions
added mobile responsive for related posts
Screenshot (206)

**Comment by vamsidhar-914** [comment]
@krishnaacharyaa  ?

**Comment by krishnaacharyaa** [comment]
Super @vamsidhar-914 
Thank you for the efforts, really appreciated.

Can you use 
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" width="512" height="512" x="0" y="0" viewBox="0 0 492.004 492.004" style="enable-background:new 0 0 512 512" xml:space="preserve"><g><path d="M382.678 226.804 163.73 7.86C158.666 2.792 151.906 0 144.698 0s-13.968 2.792-19.032 7.86l-16.124 16.12c-10.492 10.504-10.492 27.576 0 38.064L293.398 245.9l-184.06 184.06c-5.064 5.068-7.86 11.824-7.86 19.028 0 7.212 2.796 13.968 7.86 19.04l16.124 16.116c5.068 5.068 11.824 7.86 19.032 7.86s13.968-2.792 19.032-7.86L382.678 265c5.076-5.084 7.864-11.872 7.848-19.088.016-7.244-2.772-14.028-7.848-19.108z" fill="#000000" opacity="1" data-original="#000000"></path></g></svg>

For the arrow right svg
May be this is black version, make white version of it by replacing #000000 with #ffffff and make 2 svgs and kindly handle both dark and light theme


And
image

Can you please take care of this alignment, here the see more related blogs > should be aligned to the below card like 
image

And kindly upload the photo of the shimmer/skeleton things aswell, if you feel its too much in a single PR we can take this shimmer loading skeletons in another PR i am fine with that :)

IF you could close the above 2 bugs we are good to merge this PR :)

**Comment by vamsidhar-914** [comment]
fixed svg icons for light and dark mode

added loading skeletons
Screenshot (208)

is this alignment ok? or u want any change to it

**Comment by krishnaacharyaa** [comment]
> @krishnaacharyaa
> 
> fixed svg icons for light and dark mode
> 
> added loading skeletons Screenshot (208)
> 
> is this alignment ok? or u want any change to it

Can't we have it right side as earlier 
Just the end should match

You need to use something like justify-between

**Comment by krishnaacharyaa** [comment]
Here I'm not talking about the skeletons I'm just talking about the "Related blogs" and "See all related blogs" that can we have in a single row! That would be nice
Just it should be end aligned as I marked in red line in previous comment

File: frontend/src/components/PostMobileViewCardSkeleton.tsx
Comment (krishnaacharyaa) [inline frontend/src/components/PostMobileViewCardSkeleton.tsx:1]: I guess this should go under skeletons folder, not sure, whereever all the skeletons are placed, kindly place it there,
And naming sir, please follow guidelines `related-post-card-skeleton` as other skeletons namings.

**Comment by vamsidhar-914** [comment]
done it sir @krishnaacharyaa !

**Comment by krishnaacharyaa** [comment]
Thank you sir :)

**Review by krishnaacharyaa — APPROVED** [review]
LGTM, Thank you for all the constant interest and enthusiasm and completing this task :)
Really appreciated

**Pull request merged**